pkg/zk/generators.go

```diff
@@ -37,8 +37,29 @@ func headlessSvcName(z *v1beta1.ZookeeperCluster) string {
 	return fmt.Sprintf("%s-headless", z.GetName())
 }
 
+var zkDataVolume = "data"
+
 // MakeStatefulSet return a zookeeper stateful set from the zk spec
 func MakeStatefulSet(z *v1beta1.ZookeeperCluster) *appsv1.StatefulSet {
+	extraVolumes := []v1.Volume{}
+	persistence := z.Spec.Storage.Persistence
+	pvcs := []v1.PersistentVolumeClaim{}
+	if persistence != nil {
+		pvcs = append(pvcs, v1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   zkDataVolume,
+				Labels: map[string]string{"app": z.GetName()},
+			},
+			Spec: persistence.PersistentVolumeClaimSpec,
+		})
+	} else {
+		extraVolumes = append(extraVolumes, v1.Volume{
```

Comment:
When z.Spec.Storage.Persistence is not specified by user we are defaulting it to ephemeral storage. This does **not** look right.
Please see my comment on the earlier PR:
https://github.com/pravega/zookeeper-operator/pull/64/files

Reply:
No Actually, If the user doesn't specify any value for storage in that case in **withDefaults()** method in **zookeepercluster_types.go** at line no. 159 we are handling this case and making the storage as Persistence